### PR TITLE
Use different host for uploading image

### PR DESCRIPTION
**Pull request by tomorrowkey**
Add a feature to able to set upload host.
Configuration code will be changed to set upload hostname.

```ruby
Kinu.configure do |config|
  config.host = 'example.com'
  config.upload_host = 'upload.example.com'
end
```

To achieve the feature, some classs arguments little bit changed.
Please review this 🙏  @TakatoshiMaeda @wata-gh

### CODE REVIEW

File: lib/kinu.rb [hunk 1]
```diff
@@ -12,6 +12,11 @@ def self.base_uri
     (config.ssl? ? URI::HTTPS : URI::HTTP).build(host: config.host, port: config.port)
   end
 
+  def self.base_upload_uri
+    raise "Kinu.config.upload_host is not set." if config.upload_host.empty?
```
Comment (TakatoshiMaeda) [inline lib/kinu.rb]: why default behavior not usage `host`?

File: lib/kinu/configuration.rb [hunk 2]
```diff
@@ -1,11 +1,15 @@
...
   class Configuration
-    attr_writer :host, :port, :ssl
+    attr_writer :host, :upload_host, :port, :ssl
 
     def host
       @host || ""
     end
 
+    def upload_host
+      @upload_host || host || ""
```
Comment (TakatoshiMaeda) [inline lib/kinu/configuration.rb]: The more you return nil, the more concrete it will be

File: lib/kinu.rb (hunk 1 above)
Comment (tomorrowkey) [inline lib/kinu.rb]: In the case of config.host and config.upload_host isn't set, it is raised.
The message could be not suitable. I'll tweak this.

Comment (TakatoshiMaeda) [inline lib/kinu.rb]: IMO

```ruby
if config.host.empty? || config.upload_host.empty?
  raise "Kinu.config.upload_host and Kinu.config.upload_host is not set. Please set one or the other."
end
http_class = (config.ssl? ? URI::HTTPS : URI::HTT)
http_class.build(host: config.upload_host || config.host, port: config.port)
```

File: lib/kinu/configuration.rb (hunk 2 above)
Comment (TakatoshiMaeda) [inline lib/kinu/configuration.rb]: IMO

```
def upload_host
  @upload_host || nil
end
```

I think the Configuration class will be simpler to return the setting directly

File: lib/kinu.rb (hunk 1 above)
Comment (tomorrowkey) [inline lib/kinu.rb]: changed at 05902a7

File: lib/kinu/configuration.rb (hunk 2 above)
Comment (tomorrowkey) [inline lib/kinu/configuration.rb]: Just return `@upload_host` 05902a7

**Comment by TakatoshiMaeda** [comment]
LGTM thx!

**Comment by TakatoshiMaeda** [comment]
released `v1.0.0.alpha3` https://github.com/tokubai/kinu-ruby/tree/v1.0.0.alpha3

**Pull request merged**